packages-ts/gauntlet-terra-contracts/src/lib/contracts.ts

```diff
@@ -28,109 +29,164 @@ export type TerraABI = {
   [TERRA_OPERATIONS.QUERY]: JSONSchemaType<any>
 }
 
-export type Contract = {
-  id: CONTRACT_LIST
+export abstract class Contract {
+  // Contract metadata, initialized in constructor
+  readonly id: CONTRACT_LIST
+  readonly defaultVersion: string
+  readonly dirName: string
+  readonly downloadUrl: string
+
+  // Only load bytecode & schema later if needed
   abi: TerraABI
   bytecode: string
-}
+  addresses: AccAddress[] = []
+  address: AccAddress // first address, for convenience
+
+  constructor(id, dirName, defaultVersion) {
```

Comment:
It should accept `version` instead of `defaultVersion`. `version` will be `DEFAULT_VERSION` if not provided